src/amlaidatatests/connection.py

```diff
@@ -92,6 +107,8 @@ def connection_factory(default: Optional[str] = None):
             kwargs["credentials"] = credentials
 
     connection = ibis.connect(connection_string, **kwargs)
+    if result.scheme == "bigquery":
```

Comment:
Could we collapse this into the same if statement a few lines above?

Reply:
Ah, I don't think so, I can't see a way around the fact you need credentials to create a connection which you then need to modify.

Probably the cleanest way would be to have a separate function handling bq, e.g. (editing on phone!) and put everything bq related in there.

Low priority.

def create_bq_conn():
    ...


If bigquery:
    connection= create_bq_conn()